mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -0,0 +1,67 @@
+package acctest
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+)
+
+// Hardcode the Terraform resource name -> API service name mapping temporarily.
+// TODO: [tgc] read the mapping from the resource metadata files.
+var ApiServiceNames = map[string]string{
+	"google_compute_instance": "compute.googleapis.com",
+	"google_project":          "cloudresourcemanager.googleapis.com",
+}
+
+// Gets the test metadata for tgc:
+//   - test config
+//   - cai asset name
+//     For example: //compute.googleapis.com/projects/ci-test-188019/zones/us-central1-a/instances/tf-test-mi3fqaucf8
+func GetTestMetadataForTgc(service, resourceType, resourceName, config string) resource.TestCheckFunc {
```

Comment:
the service is no longer necessary here

```suggestion
func GetTestMetadataForTgc(resourceType, resourceName, config string) resource.TestCheckFunc {
```

Reply:
The acceptance test name is also used for the tgc tests. `service` is logged and will be used to put the tgc tests  into specific service packages.

Reply:
Ah, that makes sense!

If this should be the service package, it could be inferred based on the service package containing this resource's definition instead of passed in as part of the call...

Unless the plan is to automatically add the calls to this function to all VCR tests (and the mechanism adding the calls will have access to that information)?

Reply:
I think this would likely be better handle with automation but we can always decide to change it later if it causes problems.